### PR TITLE
feat(SearchResultsByCategory): Group all vehicle results into a single category

### PR DESCRIPTION
**Pull request by KaylaBrady**
ticket: https://app.asana.com/0/1200180014510248/1205197487452691/f

This PR disconnects the search results from the filters, so all results are always shown. The results will be reconnected to the new filter as part of https://app.asana.com/0/1200180014510248/1205198429752070/f

### CODE REVIEW

File: assets/src/models/searchQuery.ts [hunk 1]
```diff
@@ -9,15 +9,23 @@ export const searchPropertyDisplayConfig = {
 
 export type SearchProperty = keyof typeof searchPropertyDisplayConfig
 
+export type VehiclePropertyQuery = "all" | "vehicle" | "run" | "operator"
+export type SearchPropertyQuery = "all" | VehiclePropertyQuery | "location"
```
Comment (KaylaBrady) [inline assets/src/models/searchQuery.ts:13]: I struggled with naming here. I think we'll want 3 different types: 
* individual properties excluding all (currently `SearchProperty`)
* all possible vehicle field queries including all (`VehiclePropertyQuery`)
* all possible vehicle + location field queries (`SearchPropertyQuery`)

Maybe renaming `SearchProperty` something like `SearchFields` would convey that those are the individual fields that can be queried, and the other two can become `VehicleProperty` and `SearchProperty`? That doesn't quite feel right to me either - any alternatives come to mind?

File: assets/tests/state/searchPageState.test.ts [hunk 2]
```diff
@@ -142,14 +158,32 @@ describe("reducer", () => {
...
+        query: emptySearchQueryFactory.build({
+          text: "123",
+          categoryResultLimits: { location: 10, vehicle: 20 },
+        }),
+      })
+
+      const newState = reducer(oldState, setCategoryMatchLimit("location", 15))
+      expect(newState.query.categoryResultLimits).toEqual({
+        location: 15,
+
```
Comment (lemald) [inline assets/tests/state/searchPageState.test.ts:181]: **nitpick:** blank line (surprised `prettier` didn't catch this>).

**Review by lemald — APPROVED** [review]
Okay, one minor syntax nit but overall LGTM.

File: assets/tests/components/mapPage/searchResultsByCategory.test.tsx [hunk 3]
```diff
@@ -241,26 +206,14 @@ describe("searchResultsByProperty", () => {
...
 
-  test("For sections that have more matches, includes a 'Show more' button", () => {
-    ;(useSearchResultsByProperty as jest.Mock).mockReturnValue({
-      run: {
-        ok: {
-          matches: [runMatch],
-          hasMoreMatches: false,
-        },
-      },
+  test("When there are more vehicl matches, includes a 'Show more' button which updates the vehicle result limit on click", async () => {
```
Comment (firestack) [inline assets/tests/components/mapPage/searchResultsByCategory.test.tsx:209]:
```suggestion
  test("When there are more vehicle matches, includes a 'Show more' button which updates the vehicle result limit on click", async () => {
```

File: assets/src/models/searchQuery.ts (hunk 1 above)
Comment (firestack) [reply assets/src/models/searchQuery.ts:13]: I don't have any objections, so far I don't have any ideas for how it should be named.

**Pull request merged**